face_detection.py

```diff
@@ -0,0 +1,190 @@
+import torch
+import numpy as np
+import os
+from PIL import Image, ImageDraw
+import cv2
+from facenet_pytorch.models.mtcnn import fixed_image_standardization
+from matplotlib import pyplot as plt
+from facenet_pytorch import MTCNN, InceptionResnetV1
+import torchvision
+from torchvision import transforms
+from tqdm import tqdm
+from tkinter import Tk, filedialog
+import subprocess
+import platform
+from datetime import datetime
+from pathlib import Path
+import pandas as pd
+from smile_net import Net
+
+
+class FaceObject:
+    # center is numpy array, to present the last center point [x,y]
+    def __init__(self, box: list, pred: int, idx: int, v_len: int):
+        self.current_center = np.array([(box[0] + box[2])/2, (box[1] + box[3])/2])
+        self.predictions_in_frames = [None] * v_len
+        self.predictions_in_frames[idx] = pred
+        self.coordinates_in_frames = [None] * v_len
+        self.coordinates_in_frames[idx] = box
+
+
+def assign(face_objects_list: list, boxes, predictions, idx, v_len) -> list:
+    max_assign_distance = 50
```

Comment:
it is customary to place constants (a = 5, b = 'hello', etc) either in the top of file or in another `consts.py` file and import from there.

also - const variable names are uppercase 

--> MAX_ASSIGN_DISTANCE = 50